bciers/apps/reporting/src/data/jsonSchema/additionalReportingData/additionalReportingData.ts

```diff
@@ -10,107 +10,160 @@ export const additionalReportingDataSchema: RJSFSchema = {
-                    type: "number",
-                    title: "Emissions (t) captured for off-site transfer",
+                {
+                  if: {
+                    properties: {
+                      capture_type: {
+                        contains: { enum: ["Off-site transfer"] },
+                      },
+                    },
+                  },
+                  then: {
+                    properties: {
+                      emissions_off_site_transfer: {
+                        type: "number",
+                        title: "Emissions (t) captured for off-site transfer",
+                      },
+                    },
                   },
                 },
-              },
+              ],
             },
           ],
         },
-      ],
+      },
     },
   },
 };
 
 export const additionalReportingDataUiSchema = {
   "ui:FieldTemplate": FieldTemplate,
   "ui:classNames": "form-heading-label",
-  additional_data: { "ui:FieldTemplate": SectionFieldTemplate },
-  purpose_note: {
-    "ui:FieldTemplate": TitleOnlyFieldTemplate,
-    "ui:title": CapturedEmmissionsInfo,
-  },
-  capture_emissions: {
-    "ui:title": "Did you capture emissions?",
-    "ui:widget": RadioWidget,
+
+  captured_emissions_section: {
+    "ui:FieldTemplate": SectionFieldTemplate,
+    "ui:options": { label: false },
+    "ui:order": [
+      "purpose_note",
+      "capture_emissions",
+      "capture_type",
+      "emissions_on_site_use",
+      "emissions_on_site_sequestration",
+      "emissions_off_site_transfer",
+    ],
+    purpose_note: {
+      "ui:FieldTemplate": TitleOnlyFieldTemplate,
+      "ui:title": CapturedEmmissionsInfo,
+    },
+    capture_emissions: {
+      "ui:widget": RadioWidget,
+    },
+    capture_type: {
+      "ui:widget": multiSelectWidget,
+      "ui:options": { style: { width: "100%", textAlign: "justify" } },
+      "ui:placeholder": "Capture type",
+    },
   },
-  capture_type: {
-    "ui:widget": multiSelectWidget,
-    "ui:options": { style: { width: "100%", textAlign: "justify" } },
-    "ui:placeholder": "Capture type",
+
+  additional_data_section: {
+    "ui:FieldTemplate": SectionFieldTemplate,
+    "ui:order": ["electricity_generated"],
+    electricity_generated: {
+      "ui:placeholder": "Enter amount of electricity generated",
+    },
   },
 };
+
+/**
+ * Appends the electricity_generated field to the schema if includeElectricityGenerated is true.
+ *
+ * @returns The modified schema or the original schema if the flag is false.
+ */
+export function getUpdatedSchema(): RJSFSchema {
+  return {
+    ...additionalReportingDataSchema,
+    properties: {
+      ...additionalReportingDataSchema.properties,
+      additional_data_section: {
+        type: "object",
+        title: "Additional data",
+        properties: {
+          electricity_generated: {
+            type: "string",
+            title: "Electricity Generated",
+          },
+        },
+      },
+    },
+  };
+}
```

Comment:
we can export this as a constant and compute it only once (I'm sure it's not huge, but there's definitely a performance penalty for destructuring dictionaries).
Also, we should make functions pure where we can, it's good coding practice
```ts
function buildSchemaWithElectricityGenerated(additionalReportingDataUiScheam: RJSFSchema){
...return the logic 
}

export const additionalReportingDataWithElectricityGeneratedUiSchema : RJSFSchema = 
    buildSchemaWithElectricityGenerated(additionalReportingDataUiSchema)
```